JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts

```diff
@@ -67,15 +77,15 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
             return name;
         }
 
-        public static sanitizeString(value) {
+        public static sanitizeString(value, maxLength: number = DataSanitizer.MAX_STRING_LENGTH) {
             if (value) {
                 value = Util.trim(value);
-                if (value.toString().length > DataSanitizer.MAX_STRING_LENGTH) {
-                    value = value.toString().substring(0, DataSanitizer.MAX_STRING_LENGTH);
+                if (value.toString().length > maxLength) {
+                    value = value.toString().substring(0, maxLength);
                     _InternalLogging.throwInternal(
                         LoggingSeverity.WARNING,
                         _InternalMessageId.StringValueTooLong,
-                        "string value is too long. It has been truncated to " + DataSanitizer.MAX_STRING_LENGTH + " characters.",
+                        "string value is too long. It has been truncated to " + maxLength + " characters.",
```

Comment:
as a future task, can we possibly also get the *key* corresponding of the value that was trimmed in this message?  the message tells you what the *value* was that was truncated, but knowing that it was a custom property with name `x` or the field `url` or whatever would be really useful to track down what the source of that value was.

Reply:
It is possible to do this now with the associated length (value is different for name vs. custom property). we use this method for these two types only).